test/runtests.jl

```diff
@@ -91,6 +91,18 @@ for xtype in [:Float64, :BigFloat, :Int64]
         end
     end
 end
+
+# Treat mod separately because of discontinuities at integers
+derivs = DiffRules.diffrule(:Base, :mod, :x, :y)
+@eval begin
+    let
+        x = randn()
+        y = randn()
+        dx, dy = $(derivs[1]), $(derivs[2])
+        @test isapprox(dx, finitediff(z -> mod(z, y), float(x)), rtol=0.05)
+        @test isapprox(dy, finitediff(z -> mod(x, z), float(y)), rtol=0.05)
+    end
+end
```

Comment:
IMO it would be preferable to integrate this with the other binary tests above, similar to how different functions and e.g. `mod2pi` are handled in the unary tests.

Reply:
What the `mod2pi` tests are doing is shifting tested value away from the discontinuity so that the finite diff doesn't blow up. This is essentially what I am doing here by sampling `randn` values, which are not likely to be integers.